### PR TITLE
try win

### PR DESCRIPTION
**Pull request by ocefpaf**
Compile on Windows.

### CODE REVIEW

**Comment by ocefpaf** [comment]
It seems that it does not like `vc 14`

```shell
-- Installing: C:/bld/libkml_1498939036686/_b_env/Library/include/uriparser/UriIp4.h
[ 11%] Completed 'URIPARSER'
[ 11%] Built target URIPARSER
Scanning dependencies of target kmlbase
[ 12%] Building CXX object src/kml/base/CMakeFiles/kmlbase.dir/attributes.cc.obj
attributes.cc
[ 13%] Building CXX object src/kml/base/CMakeFiles/kmlbase.dir/csv_splitter.cc.obj
csv_splitter.cc
[ 13%] Building CXX object src/kml/base/CMakeFiles/kmlbase.dir/date_time.cc.obj
date_time.cc
[ 14%] Building CXX object src/kml/base/CMakeFiles/kmlbase.dir/expat_handler_ns.cc.obj
expat_handler_ns.cc
[ 15%] Building CXX object src/kml/base/CMakeFiles/kmlbase.dir/expat_parser.cc.obj
expat_parser.cc
[ 16%] Building CXX object src/kml/base/CMakeFiles/kmlbase.dir/file.cc.obj
file.cc
[ 16%] Building CXX object src/kml/base/CMakeFiles/kmlbase.dir/file_win32.cc.obj
file_win32.cc
C:\bld\libkml_1498939036686\work\libkml-1.3.0\src\kml\base\file_win32.cc(45): error C2137: empty character constant
NMAKE : fatal error U1077: 'C:\PROGRA~2\MI0E91~1.0\VC\bin\cl.exe' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\nmake.exe"' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\nmake.exe"' : return code '0x2'
Stop.
(C:\bld\libkml_1498939036686\_b_env) C:\bld\libkml_1498939036686\work\libkml-1.3.0\build>if errorlevel 1 exit 1 
```

**Comment by ocefpaf** [comment]
@gillins do you think this has an easy fix or should I just skip the `vc 14`?

**Comment by gillins** [comment]
See "Visual Studio 2015 notes" down the bottom of this page: https://trac.osgeo.org/gdal/wiki/LibKML#Buildwithlibkml

**Comment by gillins** [comment]
Below is a patch for `vc14`. Let me know if this doesn't fix it and I'll delve more deeply. Not sure why they didn't put and ifdef or something around this.

```
--- src/kml/base/file_win32.cc.orig	2017-07-03 10:01:57.585684177 +1000
+++ src/kml/base/file_win32.cc	2017-07-03 10:02:12.705684137 +1000
@@ -42,7 +42,7 @@
 // Internal to the win32 file class. We need a conversion from string to
 // LPCWSTR.
 static std::wstring Str2Wstr(const string& str) {
-  std::wstring wstr(str.length(), L'');
+  std::wstring wstr(str.length(), L'\0');
   std::copy(str.begin(), str.end(), wstr.begin());
   return wstr;
 }
```

**Comment by ocefpaf** [comment]
> Not sure why they didn't put and ifdef or something around this.

The project is hosted on GitHub so, if this works, we can send it upstream.

Testing it now...

PS: I had https://trac.osgeo.org/gdal/wiki/LibKML#Buildwithlibkml open in my browser and missed that :flushed:

File: recipe/empty_char.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- libkml-1.3.0.orig/src/kml/base/file_win32.cc	2015-12-21 15:23:05.000000000 -0200
++++ libkml-1.3.0/src/kml/base/file_win32.cc	2017-07-03 08:56:24.542091159 -0300
+@@ -42,7 +42,7 @@
+ // Internal to the win32 file class. We need a conversion from string to
+ // LPCWSTR.
+ static std::wstring Str2Wstr(const string& str) {
+-  std::wstring wstr(str.length(), L'');
++  std::wstring wstr(str.length(), L'\0');
```
Comment (ocefpaf) [inline recipe/empty_char.patch:8]: @gillins it worked!

**Pull request merged**